src/Builder.js

```diff
@@ -57,31 +99,76 @@ Builder.prototype = {
   },
 
   star: function(expr) {
+    if (!(expr instanceof pexprs.PExpr)) {
+      expr = this.fromRecipe(expr);
+    }
     return new pexprs.Star(expr);
   },
 
   plus: function(expr) {
+    if (!(expr instanceof pexprs.PExpr)) {
+      expr = this.fromRecipe(expr);
+    }
     return new pexprs.Plus(expr);
   },
 
   opt: function(expr) {
+    if (!(expr instanceof pexprs.PExpr)) {
+      expr = this.fromRecipe(expr);
+    }
     return new pexprs.Opt(expr);
   },
 
   not: function(expr) {
+    if (!(expr instanceof pexprs.PExpr)) {
+      expr = this.fromRecipe(expr);
+    }
     return new pexprs.Not(expr);
   },
 
   la: function(expr) {
+    // TODO: temporary to still be able to read old recipes
+    return this.lookahead(expr);
+  },
+
+  lookahead: function(expr) {
+    if (!(expr instanceof pexprs.PExpr)) {
+      expr = this.fromRecipe(expr);
+    }
     return new pexprs.Lookahead(expr);
   },
 
   lex: function(expr) {
+    if (!(expr instanceof pexprs.PExpr)) {
+      expr = this.fromRecipe(expr);
+    }
     return new pexprs.Lex(expr);
   },
 
   app: function(ruleName, optParams) {
+    if (optParams && optParams.length > 0) {
+      optParams = optParams.map(function(param) {
+        return param instanceof pexprs.PExpr ? param :
+          this.fromRecipe(param);
+      }, this);
+    }
     return new pexprs.Apply(ruleName, optParams);
+  },
+
+  fromRecipe: function(recipe) {
+    // the meta-info of 'grammar' is proccessed in Builder.grammar
+    var res = this[recipe[0]].apply(this,
```

Comment:
Is `res` should for `result` here? It took me a while to figure out -- I kept thinking it meant "resource" or something. It's a minor point, but I would either call this "result" or "ans", which are the two most common names we use elsewhere in Ohm.